instruments/tests/test_tektronix/test_tekdpo4104.py

```diff
@@ -0,0 +1,503 @@
+        inst.aquisition_running = False
+        assert not inst.aquisition_running
+        inst.aquisition_running = True
+
+
+def test_aquisition_continuous():
+    """Get / set status of aquisition continuous."""
+    with expected_protocol(
+            ik.tektronix.TekDPO4104,
+            [
+                "ACQ:STOPA?",
+                "ACQ:STOPA SEQ",
+                "ACQ:STOPA?",
+                "ACQ:STOPA RUNST"
+            ],
+            [
+                "RUNST",
+                "SEQ"
+            ]
+    ) as inst:
+        assert inst.aquisition_continuous
+        inst.aquisition_continuous = False
+        assert not inst.aquisition_continuous
+        inst.aquisition_continuous = True
+
+
+@pytest.mark.parametrize("data_width", (1, 2))
+def test_data_width(data_width):
+    """Get / set data width with valid values."""
+    with expected_protocol(
+            ik.tektronix.TekDPO4104,
+            [
+                f"DATA:WIDTH {data_width}",
+                "DATA:WIDTH?",
+            ],
+            [
+                f"{data_width}"
+            ]
+    ) as inst:
+        inst.data_width = data_width
+        assert inst.data_width == data_width
+
+
+def test_data_width_out_of_range():
+    """Raise Value Error if input value is out of range."""
+    with expected_protocol(
+            ik.tektronix.TekDPO4104,
+            [
+            ],
+            [
+            ]
+    ) as inst:
+        with pytest.raises(ValueError) as exc_info:
+            inst.data_width = 42
+        exc_msg = exc_info.value.args[0]
+        assert exc_msg == "Only one or two byte-width is supported."
+
+
+@given(offset=st.floats(min_value=-100, max_value=100))
+def test_y_offset(offset):
+    """Get / set Y offset of currently selected data source."""
+    with expected_protocol(
+            ik.tektronix.TekDPO4104,
+            [
+                f"WFMP:YOF {offset}",
+                "WFMP:YOF?"
+            ],
+            [
+                f"{offset}"
+            ]
+    ) as inst:
+        inst.y_offset = offset
+        assert inst.y_offset == offset
+
+
+def test_force_trigger():
+    """Force a trigger event to occur."""
+    with expected_protocol(
+            ik.tektronix.TekDPO4104,
+            [
+                "TRIG FORCE"
+            ],
+            [
+            ]
+    ) as inst:
+        inst.force_trigger()
+
+
+# CHANNELS #
+
+
+channels_to_try = range(4)
+channels_to_try_ids = [f"CH{it}" for it in channels_to_try]
+
+
+@pytest.mark.parametrize("channel", channels_to_try, ids=channels_to_try_ids)
+def test_channel_init(channel):
+    """Initialize a channel."""
+    with expected_protocol(
+            ik.tektronix.TekDPO4104,
+            [
+            ],
+            [
+            ]
+    ) as inst:
+        assert inst.channel[channel]._idx == channel + 1
+
+
+@pytest.mark.parametrize("channel", channels_to_try, ids=channels_to_try_ids)
+@given(coupling=st.sampled_from(ik.tektronix.TekDPO4104.Coupling))
```

Comment:
I haven't tried this, but can you pytest parametrize over an enum?

Reply:
Just tried it out for fun, in fact you can :) Here would be:
```python
@pytest.mark.parametrize("coupling", ik.tektronix.TekDPO4104.Coupling)
```

That would probably make more sense than sampling with hypothesis.